maize/building-and-booting.rst

```diff
@@ -0,0 +1,186 @@
+To save network bandwidth, we have a copy of the kernel's Git repo on Maize already, which you should clone.
+
+.. code:: sh
+
+    # Clone the repo. This will clone it to the current working directory, as a
+    # directory named linux. You can clone it to a different directory or move
+    # it after cloning it if you would prefer it to be elsewhere.
+    myx500@cs-u-maize.cs.umn.edu:~$ git clone /srv/git/linux
+
+    # Once the kernel is cloned, cd into it before proceeding to the following
+    # steps.
+    myx500@cs-u-maize.cs.umn.edu:~$ cd linux
+
+If, when cloning the repo, you get an error like the following, run the command from the error:
+
+.. code:: sh
+
+    myx500@cs-u-maize.cs.umn.edu:~$ git clone /srv/git/linux
+    Cloning into 'linux'...
+    fatal: detected dubious ownership in repository at '/srv/git/linux/.git'
+    To add an exception for this directory, call:
+
+            git config --global --add safe.directory /srv/git/linux/.git
+    fatal: Could not read from remote repository.
+
+    Please make sure you have the correct access rights
+    and the repository exists.
+
+    myx500@cs-u-maize.cs.umn.edu:~$ git config --global --add safe.directory /srv/git/linux/.git
+
+    # After running the command, re-run git clone.
+    myx500@cs-u-maize.cs.umn.edu:~$ git clone /srv/git/linux
+
+We're using a tool called `virtme-ng <https://github.com/arighi/virtme-ng>`_ to automate the process of building the kernel and starting a VM containing it.
+Each of these two steps is a single command.
+
+.. code:: sh
+
+    # Make sure that you've cd'd into the kernel repo. This command builds the
+    # kernel, using a minimal default configuration that will build quickly and
+    # work for these steps. Later, we'll see how you can adjust the
+    # configuration.
+    myx500@cs-u-maize.cs.umn.edu:~/linux$ vng --verbose --build
+
+    # This command boots a virtual machine that has access to your home
+    # directory.
+    myx500@cs-u-maize.cs.umn.edu:~/linux$ vng --verbose
+
+After launching the virtual machine, you should be in the same directory, but the hostname should display as ``virtme-ng``.
+
+.. code:: sh
+
+    # This command is now running in the virtual machine.
+    myx500@virtme-ng:~/linux$ uname -a
+    Linux virtme-ng 6.12.0-rc1-virtme #1 SMP PREEMPT_DYNAMIC Sat Oct  5 19:44:31 CDT 2024 x86_64 x86_64 x86_64 GNU/Linux
+
+You can exit out of the virtual machine by exiting the shell, either with the ``exit`` shell built-in or with the :kbd:`Ctrl+D` keypress [#ctrld]_.
+
+Now that we have the kernel working, we can see how to modify and configure it.
+In this tutorial, we won't cover the process of authoring a new kernel module.
+Instead, we'll use a "Hello, World!" module that's already been written.
+This module is present on a branch in the kernel repo on Maize.
+
+Check out the branch with the module on it.
+
+.. code:: sh
+
+    myx500@cs-u-maize.cs.umn.edu:~/linux$ git checkout hello_module
```

Comment:
Thoughts on using `switch` instead of `checkout`?

Reply:
No opinion; I use checkout but I don't have a strong preference for which we teach. I'll change it with the other changes.

Reply:
Done in 99c6f6c60fe7da0057b8d5c1c94af3d2edd3d74f